lib/BillCalculator.rb

```diff
@@ -0,0 +1,54 @@
+class BillCalculator
+  attr_accessor :total, :order
+  attr_writer :meal_options
+
+    def initialize(meal_options)
+      @meal_options = meal_options
+      @total = 0
+      @order = []
+    end
+
+    def calculate_total(dishes)
+      dishes.each do |k,v|
+        if k == @meal_options.keys[0]
+          @total += v * @meal_options.values[0]
+        elsif k == @meal_options.keys[1]
+          @total += v * @meal_options.values[1]
+        elsif k == @meal_options.keys[2]
+          @total += v * @meal_options.values[2]
+        elsif k == @meal_options.keys[3]
+          @total += v * @meal_options.values[3]
+        elsif k == @meal_options.keys[4]
+          @total += v * @meal_options.values[4]
+        elsif k == @meal_options.keys[5]
+          @total += v * @meal_options.values[5]
+        end
```

Comment:
With hashes, you can look up values by the key, so for example:
```
inventory = {:pencil => 5, :notebook => 6, :envelope => 3}

p inventory[:pencil] // prints out 5
```

Can you see how this hash lookup can be used to simplify this code? Are there other places the code can be refactored in this way to improve readability?